lib/request-parser.js

```diff
@@ -7,10 +7,75 @@ const bodyParser = require("body-parser");
 const cookieParser = require("cookie-parser");
 const multer = require("multer");
 const tmp = require("tmp");
+const path = require("path");
+const util = require("./helpers/util");
 
 // Clean-up the temp directory, even if the app crashes
 tmp.setGracefulCleanup();
 
+/**
+ * Generates a middleware that parses multipart/form-data
+ */
+function generateMultipartFormDataMiddleware (options) {
+  options.allowAll = options.allowAll === undefined ? true : options.allowAll;
+  const uploader = multer(options);
+  return function multipartFormData (req, res, next) {
+    // Compatibility with old multer
+    req.body = req.body || {};
+    req.files = req.files || {};
+
+    // Get all the "file" params
+    if (util.isSwaggerRequest(req) && req.swagger.params.length > 0) {
+      const fileFields = [];
+      req.swagger.params.forEach((param) => {
+        if (param.in === "formData" && param.type === "file") {
+          fileFields.push({
+            name: param.name,
+            maxCount: 1
+          });
+        }
+      });
+
+      uploader.fields(fileFields)(req, res, (err) => {
+        if (err && err.code === "LIMIT_UNEXPECTED_FILE") {
+          next(); // let request-validator handle
+        }
+        else {
+          next(err);
+        }
+      });
+    }
+    else {
+      // Handle the multipart/form-data (even if it doesn't have any file fields)
+      uploader.any()(req, res, next);
+    }
+  };
+}
+
+function multerCompatability (req, res, next) {
+  function standardize (file) {
+    if (file.originalname) {
+      file.extension = path.extname(file.originalname).slice(1);
+    }
+    if (file.filename) {
+      file.name = file.filename;
+    }
+  }
+
+  if (req.files) {
+    if (req.files.length > 0) {
+      req.files.forEach((file) => standardize(file));
+    }
+    if (Object.keys(req.files).length > 0) {
+      Object.keys(req.files).forEach((filekey) => {
+        const filearray = Array.from(req.files[filekey]);
```

Comment:
If they are specified fields, they will generally come in as named properties on an object and will sometimes be an array, so if we standardize on looping over an array for standardization we avoid some more complex scenarios.